File: trunk/Som/Topology/ITopology.cs
using System.Collections.Generic;
using Som.Network;

namespace Som.Topology
{
    public interface ITopology
    {
        int NeuronsCount { get; }

        int RowCount { get; }

        int ColCount { get; }
        
        double WholeTopologyRadius { get; }
        double[] DistancesToWinner { get; }

        int GetNeuronNumber(Location location);
        
        Location GetNeuronLocation(int neuronNumber);

        IList<int> GetDirectlyConnectedNeurons(int neuronNumber);

        List<int> GetNeuronsInRadius(int neuronNumber, double radius);

        bool Overlaps(int neuronA, int neuronB, double radius);
    }
}

File: trunk/JavaSom/Topology/ITopology.java
package Topology;

	public interface ITopology
	{
		int getNeuronsCount();
		int getRowCount();
		int getColCount();
		double getWholeTopologyRadius();
		java.util.List<Integer> GetDirectlyConnectedNeurons(int neuronNumber);
		java.util.HashMap<Integer, Double> GetNeuronsInRadius(int neuronNumber, double radius);
		boolean Overlaps(int neuronA, int neuronB, double radius);
	}